### PR TITLE
docs: Add Agile Development pillars

### LINKED ISSUE #630: Adding Delivery Plan as part of the Agile Process
Child to this parent: https://github.com/microsoft/code-with-engineering-playbook/issues/620

### PR DESCRIPTION
**Pull request by janine-zhu**
Closes #630  -  Adding Delivery Plan to the Agile Playbook

1. Added Delivery Plan bullet to Agile Development Readme 
2. Added new folder delivery-plan and README.md about delivery-plan

### CODE REVIEW

**Review by jskulavik — CHANGES_REQUESTED** [review]
I've pushed requested changes to our common fork @janine-zhu.

File: docs/agile-development/delivery-plan/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## Approach
+
+One approach you can take to accomplish is with stickies and a spreadsheet.
+
+Step 1: Stack rank the features for EVERYTHING in the backlog; including
+
+- Functional Features
+
+- [Non-functional Features](docs/TECH-LEADS-CHECKLIST.md)
```
Comment (jskulavik) [inline docs/agile-development/delivery-plan/README.md:25]: Remove this line, it's a dead link and out of context

File: docs/agile-development/README.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## Sections
 
+- [Delivery Plans](delivery-plan/readme.md)
```
Comment (fnocera) [inline docs/agile-development/README.md:5]: I would recommend putting the sections in alphabetical order so shift down

File: docs/agile-development/delivery-plan/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Step 2: T-Shirt features in terms of working weeks per person. In some scenarios, you have no idea how complex the work may be. In this situation, you can ask for time to conduct a spike (timebox the effort spend on spikes).
+
+Step 3: Calculate the capacity for the team based on the number of weeks person with his/her start and end date and minus holidays, vacation, conferences, training, and onboarding days. Also, minus time if the person is also working on defects and support.
+
+Step 4: Based on your capacity, you now have additional options:
+
+- Ask for more resources. Caution: onboarding new resources takes time.
+
+- Reduce the scope to the most MVP.  Caution: as you trim more of the scope, you risk negatively impacting the project's value propositions. Consider a cupcake which is everything you need. You don't want to skim on the frosting.
```
Comment (fnocera) [inline docs/agile-development/delivery-plan/README.md:40]:
```suggestion
- Reduce the scope to an MVP.  Caution: as you trim more of the scope, you risk negatively impacting the project's value propositions. Consider a cupcake which is everything you need. You don't want to skim on the frosting.
```

File: docs/agile-development/delivery-plan/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# Delivery Plan
+
+## Goals
+
+While Scrum does not require, and in fact discourages planning more than one sprint at a time, most of us work in enterprises where we are dependent on each other. For example: marketing, sales, and support needs to have know what is coming up so they can make plans for budget, people and resources. A delivery plan helps to ensure your teams are aligning with your organizational goals.
```
Comment (fnocera) [inline docs/agile-development/delivery-plan/README.md:5]: Nit: would suggest using "Agile" vs "Scrum" as different teams tend to use different versions of Agile, not just Scrum
```suggestion
While Agile does not require, and in fact discourages planning more than one sprint at a time, most of us work in enterprises where we are dependent on each other. For example: marketing, sales, and support needs to have know what is coming up so they can make plans for budget, people and resources. A delivery plan helps to ensure your teams are aligning with your organizational goals.
```

File: docs/agile-development/delivery-plan/README.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+Step 3: Calculate the capacity for the team based on the number of weeks person with his/her start and end date and minus holidays, vacation, conferences, training, and onboarding days. Also, minus time if the person is also working on defects and support.
+
+Step 4: Based on your capacity, you now have additional options:
+
+- Ask for more resources. Caution: onboarding new resources takes time.
+
+- Reduce the scope to the most MVP.  Caution: as you trim more of the scope, you risk negatively impacting the project's value propositions. Consider a cupcake which is everything you need. You don't want to skim on the frosting.
+
+- Ask for more time. Usually, this is the most flexible, but if there is a marketing date that you need to hit, this might be a flexiblity point.
```
Comment (fnocera) [inline docs/agile-development/delivery-plan/README.md:42]: Suggest making the marketing date an example as it's not something we deal with very often in CSE. 
```suggestion
- Ask for more time. Usually, this is the most flexible, but depends on the customer (there could be a fixed marketing date for a new product).
```

**Review by fnocera — APPROVED** [review]
Thanks for the contribution @janine-zhu.

cc: @meulta to raise visibility of this PR

File: docs/agile-development/delivery-plan/README.md (hunk 1 above)
Comment (TessFerrandez) [reply docs/agile-development/delivery-plan/README.md:25]:
```suggestion
```

**Review by TessFerrandez — APPROVED** [review]

**Review by meulta — CHANGES_REQUESTED** [review]
@janine-zhu thanks a lot for the PR. Sorry it slipped through the cracks.

Can you please take a look at the comments and make changes where relevant before we can merge it?

Thank you!

**Comment by TessFerrandez** [comment]
@jskulavik and @janine-zhu -- are the requested changes implemented so we can merge? -- currently blocking on a request for changes from May 12

File: docs/agile-development/README.md [hunk 6]
```diff
@@ -2,8 +2,10 @@
...
 - [Backlog Management](backlog-management/README.md)
 - [Collaboration](collaboration/README.md)
+- [Delivery Plan](delivery-plan/readme.md)
```
Comment (TessFerrandez) [inline docs/agile-development/README.md:8]:
```suggestion
- [Delivery Plan](delivery-plan/README.md)
```

Comment (TessFerrandez) [reply docs/agile-development/README.md:8]: casing matters for both mkdocs and linters

**Review by TessFerrandez — APPROVED** [review]

File: docs/agile-development/README.md [hunk 7]
```diff
@@ -2,8 +2,10 @@
 
 ## Sections
 
+- [Delivery Plans](delivery-plan/readme.md)
 - [Backlog Management](backlog-management/README.md)
 - [Collaboration](collaboration/README.md)
+- [Delivery Plan](delivery-plan/README.md)
```
Comment (TessFerrandez) [inline docs/agile-development/README.md:8]:
```suggestion
```

Comment (TessFerrandez) [reply docs/agile-development/README.md:8]: already present above

File: docs/agile-development/README.md [hunk 8]
```diff
@@ -2,8 +2,10 @@
 
 ## Sections
 
+- [Delivery Plans](delivery-plan/readme.md)
```
Comment (TessFerrandez) [inline docs/agile-development/README.md:5]:
```suggestion
- [Delivery Plan](delivery-plan/readme.md)
```

**Review by TessFerrandez — APPROVED** [review]
Some quick language (and duplicated links) fixes

File: docs/agile-development/README.md [hunk 9]
```diff
@@ -2,6 +2,7 @@
 
 ## Sections
 
+- [Delivery Plan](delivery-plan/readme.md)
 - [Backlog Management](backlog-management/README.md)
 - [Collaboration](collaboration/README.md)
 - [Retrospectives](retrospectives.md)
```
Comment (TessFerrandez) [inline docs/agile-development/README.md:8]:
```suggestion
- [Backlog Management](backlog-management/README.md)
- [Collaboration](collaboration/README.md)
- [Delivery Plan](delivery-plan/README.md)
- [Retrospectives](retrospectives.md)
```

**Review by TessFerrandez — APPROVED** [review]
Fixing link to readme

**Pull request merged**